[addons][audioengine] cleanup the addon audioengine to match "C", match clang and to make more safe.

AlwinEsch opened the pull request:
## Description

Cleanup the addon audioengine to match "C", match clang and to make more safe.

**The API change only on audioengine where only be used on webbrowser addon**

Related documentation http://alwinesch.github.io/group__cpp__kodi__audioengine.html

Commit 1:
-----------------------------
**[addons][audioengine] add `#ifdef __cplusplus`**

This to separate the C++ and C parts inside.
In following commits becomes it more cleaned to have them also
working in C

Commit 2:
-----------------------------
**[addons][audioengine] change interface code to match "C"**

This changes the code where work between Kodi and addon to correct
"C" where also allow (in future) to use in "C" only.

There on few functions the "struct" added where "C" needed, further
is a separate class added where handle AudioEngineFormat itself and
them before is renamed to AUDIO_ENGINE_FORMAT.
This was needed to remove the C++ contructor and functions inside structure
where not match to "C".

Commit 3:
-----------------------------
**[addons][audioengine] separate Kodi and Addon enum parts**

This add own enums on addon interface to identify channels and stream types.
That becomes be by 4 new functions translated between Kodi and Addon.

The enums become futher on next changes a detailed documentation to have
better on addons.

Also are this thought to use in future on AudioDecoder interface and if
comes (maybe) back on audio dsp.

Most mandatory background of this is to make 100% sure, that if on Kodi's
AudioEngine is something changed, that the addon not become a mistmatch
with them.

Commit 4:
-----------------------------
**[addons][audioengine] clang code style cleanup**

This cleanup the code to match complete the clang style.

Commit 5:
-----------------------------
**[addons][audioengine] rename CAddonAEStream to CAEStream**

This thought to reduce his string length and the "Addon" inside not needed,
this is already in a namespace kodi::audioengine and as this can nowhere
else used as on addon, can it be leaved away.

Commit 6:
-----------------------------
**[addons][audioengine] update documentation**

This cleanup and update the addon audioengine documentation.

Commit 7:
-----------------------------
**[addons][audioengine] increase API version to 1.1.0**

<!--- Provide a general summary of your change in the Pull Request title above -->
<!--- Describe your change in detail here -->

## Motivation and Context
Related to https://github.com/xbmc/xbmc/issues/17514, to have a correct "C" ABI.
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here -->

## How Has This Been Tested?
Currently only be used on webbrowser addon and runtime tested with this:
- https://github.com/AlwinEsch/web.browser.chromium/blob/master/src/addon/audio/AudioHandler.cpp

<!--- Please describe in detail how you tested your change -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc -->

## Screenshots (if appropriate):

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [X] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
<!--- Go over all the following points, and put an `X` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

AlwinEsch committed:
[addons][audioengine] add #ifdef __cplusplus

This to separate the C++ and C parts inside.
In following commits becomes it more cleaned to have them also
working in C

AlwinEsch committed:
[addons][audioengine] change interface code to match "C"

This changes the code where work between Kodi and addon to correct
"C" where also allow (in future) to use in "C" only.

There on few functions the "struct" added where "C" needed, further
is a separate class added where handle AudioEngineFormat itself and
them before is renamed to AUDIO_ENGINE_FORMAT.
This was needed to remove the C++ contructor and functions inside structure
where not match to "C".

AlwinEsch committed:
[addons][audioengine] separate Kodi and Addon enum parts

This add own enums on addon interface to identify channels and stream types.
That becomes be by 4 new functions translated between Kodi and Addon.

The enums become futher on next changes a detailed documentation to have
better on addons.

Also are this thought to use in future on AudioDecoder interface and if
comes (maybe) back on audio dsp.

Most mandatory background of this is to make 100% sure, that if on Kodi's
AudioEngine is something changed, that the addon not become a mistmatch
with them.

AlwinEsch committed:
[addons][audioengine] rename CAddonAEStream to CAEStream

This thought to reduce his string length and the "Addon" inside not needed,
this is already in a namespace kodi::audioengine and as this can nowhere
else used as on addon, can it be leaved away.

AlwinEsch committed:
[addons][audioengine] clang code style cleanup

This cleanup the code to match complete the clang style.

AlwinEsch committed:
[addons][audioengine] increase API version to 1.1.0

phunkyfish reviewed (approved):
Looks good, nice cleanup. Once you get Jenkins happy of course ;)

phunkyfish commented:
jenkins build this please

AlwinEsch committed:
[addons][doxygen] update doc views

There is a fix about " on python macros, further is the menu width increased
and the defines are not shows in summary view (looks ugly).

AlwinEsch committed:
[addons] fix CStructHdl use by copy class to class

Before has in case the "C" structure to left from Kodi by a copy from right on
addon not worked and Kodi not becomed wanted data.

Further was on blank "C" structure use no 0 set inside.

AlwinEsch committed:
[addons][audioengine] separate "C" and "C++" parts of headers

By them becomes the "C" part placed in own header to see better what language
is where and to make easier for other languages.

AlwinEsch committed:
[addons][audioengine] update documentation

This cleanup and update the addon audioengine documentation.
